### PR TITLE
Cyclomatic complexity fix

### PR DESCRIPTION
**Pull request by ooki2day**

### CODE REVIEW

File: network/host/dht.go [hunk 1]
```diff
@@ -515,6 +451,11 @@ func (dht *DHT) iterate(ctx Context, t routing.IterateType, target []byte, data
 
 		sort.Sort(routeSet)
 
+		// value, closest, err = dht.iterateIsDone(t, &queryRest, routeSet, data, ht, closestNode)
+		// if queryRest == true {
+		// 	continue
+		// }
+
```
Comment (ita-sammann) [inline network/host/dht.go]: What is it?

File: network/host/dht.go [hunk 2]
```diff
@@ -550,6 +491,129 @@ func (dht *DHT) iterate(ctx Context, t routing.IterateType, target []byte, data
...
+func (dht *DHT) sendMessageToAlphaNodes(
+	routeSet *routing.RouteSet,
+	queryRest bool,
+	t routing.IterateType,
+	ht *routing.HashTable,
+	contacted *map[string]bool,
+	target []byte,
+	futures *[]transport.Future,
+	removeFromRouteSet []*node.Node) {
```
Comment (ita-sammann) [inline network/host/dht.go]:
```
...
removeFromRouteSet []*node.Node,
) {
```
I think this way it looks cleaner

File: network/host/dht.go [hunk 3]
```diff
@@ -550,6 +491,129 @@ func (dht *DHT) iterate(ctx Context, t routing.IterateType, target []byte, data
...
+// 				future, _ := dht.transport.SendRequest(msg)
+// 				// We do not need to handle result of this message
+// 				future.Cancel()
+// 			}
+// 			return nil, nil, nil
+// 		}
+// 	} else {
+// 		closestNode = routeSet.FirstNode()
+// 	}
+// }
```
Comment (ita-sammann) [inline network/host/dht.go]: ?

File: network/host/dht.go [hunk 4]
```diff
@@ -377,176 +384,268 @@ func (dht *DHT) iterate(ctx Context, t routing.IterateType, target []byte, data
...
-				case routing.IterateFindValue:
-					responseData := result.Data.(*message.ResponseDataFindValue)
-					routeSet.Extend(routing.RouteNodesFrom(responseData.Closest))
-					if responseData.Value != nil {
-						// TODO When an iterateFindValue succeeds, the initiator must
-						// store the key/value pair at the closest receiver seen which did
-						// not return the value.
-						return responseData.Value, nil, nil
-					}
+func checkAndRefreshTieForBucket(t routing.IterateType, ht *routing.HashTable, target []byte) {
```
[the remark this answers is not in the document]
Comment (ooki2day) [reply network/host/dht.go]: yep

File: network/host/dht.go [hunk 5]
```diff
@@ -740,6 +799,63 @@ func (dht *DHT) handleMessages(start, stop chan bool) {
...
+		ctx, err = cb.SetNodeByID(msg.Receiver.ID).Build()
+	}
+	if err != nil {
+		// TODO: Do something sane with error!
+		log.Println(err) // don't return this error cuz don't know what to do with
+	}
+	return ctx
+}
+
+func (dht *DHT) switchType(ctx Context, msg *message.Message, ht *routing.HashTable) {
```
[the remark this answers is not in the document]
Comment (ooki2day) [reply network/host/dht.go]: fixed

File: network/host/dht.go [hunk 6]
```diff
@@ -740,6 +799,63 @@ func (dht *DHT) handleMessages(start, stop chan bool) {
...
+			log.Println(response.Result)
+		}
+		log.Println(response.Error)
+	case <-time.After(dht.options.MessageTimeout):
+		future.Cancel()
+		log.Println("timeout")
+	}
+}
+
+func checkRecvID(cb ContextBuilder, msg *message.Message) Context {
```
[the remark this answers is not in the document]
Comment (ooki2day) [reply network/host/dht.go]: fixed

File: network/host/dht.go [hunk 7]
```diff
@@ -377,176 +384,268 @@ func (dht *DHT) iterate(ctx Context, t routing.IterateType, target []byte, data
...
+			// table in hopes that it might come back online in the f.
+			*removeFromRouteSet = append(*removeFromRouteSet, msg.Receiver)
+			continue
 		}
+
+		*futures = append(*futures, res)
+	}
+}
+
+func checkRoutingIterateType(t routing.IterateType, messageBuilder *message.Builder, target []byte) {
```
[the remark this answers is not in the document]
Comment (ooki2day) [reply network/host/dht.go]: fixed

File: network/host/dht.go [hunk 8]
```diff
@@ -377,176 +384,268 @@ func (dht *DHT) iterate(ctx Context, t routing.IterateType, target []byte, data
...
+			return true
+		}
+	case <-time.After(dht.options.MessageTimeout):
+		close(resultChan)
+		return true
+	}
+	return false
+}
+
+func (dht *DHT) setUpResultChan(futures *[]transport.Future, ctx Context, resultChan chan *message.Message) {
```
[the remark this answers is not in the document]
Comment (ooki2day) [reply network/host/dht.go]: fixed

File: network/host/dht.go [hunk 9]
```diff
@@ -377,176 +384,268 @@ func (dht *DHT) iterate(ctx Context, t routing.IterateType, target []byte, data
...
+				return
+			case <-time.After(dht.options.MessageTimeout):
+				future.Cancel()
+				return
 			}
+		}(f)
+	}
+}
+
+func (dht *DHT) sendMessageToAlphaNodes(
```
[the remark this answers is not in the document]
Comment (ooki2day) [reply network/host/dht.go]: fixed

File: network/host/dht.go [hunk 10]
```diff
@@ -377,176 +384,268 @@ func (dht *DHT) iterate(ctx Context, t routing.IterateType, target []byte, data
...
-						return responseData.Value, nil, nil
-					}
+func checkAndRefreshTieForBucket(t routing.IterateType, ht *routing.HashTable, target []byte) {
+	if t == routing.IterateBootstrap {
+		bucket := routing.GetBucketIndexFromDifferingBit(target, ht.Origin.ID)
+		ht.ResetRefreshTimeForBucket(bucket)
+	}
+}
+
+func removeNodesFromRouteSet(routeSet *routing.RouteSet, removeFromRouteSet *[]*node.Node) {
```
[the remark this answers is not in the document]
Comment (ooki2day) [reply network/host/dht.go]: I think its a bad idea cuz RemoveMany didn't mean that it have to remove n consecutive nodes. How do you imagine "remove many pointers" func implementation?

File: network/host/dht.go [hunk 11]
```diff
@@ -328,7 +310,11 @@ func (dht *DHT) Bootstrap() error {
 	}
 
 	wg.Wait()
+	err := dht.iterateHt(cb)
+	return err
```
Comment (dmitry-panchenko) [inline network/host/dht.go]: You could actually just return `dht.iterateHt(cb)` w/o extra assignment.

File: network/host/routing/hashtable.go [hunk 12]
```diff
@@ -193,13 +193,26 @@ func (ht *HashTable) GetClosestContacts(num int, target []byte, ignoredNodes []*
...
+
+func (ht *HashTable) selectParallelCalls(leftToAdd int,
+	indexList []int,
+	ignoredNodes []*node.Node,
+	routeSet *RouteSet,
+) []int {
+	var index int
 	for leftToAdd > 0 && len(indexList) > 0 {
-		index, indexList = indexList[0], indexList[1:]
+		index, indexList = (indexList)[0], (indexList)[1:]
```
Comment (dmitry-panchenko) [inline network/host/routing/hashtable.go]: Have to remove extra braces here.

Comment (ooki2day) [reply network/host/routing/hashtable.go]: fixed

File: network/host/dht.go (hunk 11 above)
Comment (ooki2day) [reply network/host/dht.go]: fixed

**Pull request merged**